### PR TITLE
GEOMESA-2204 Fixing Kafka consumer sleep while more data to read

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  class ConsumerRunnable(val consumer: Consumer[Array[Byte], Array[Byte]], frequency: Long) extends Runnable {
+
+    private var errorCount = 0
+
+    override def run(): Unit = {
+      try {
+        while (!closed.get()) {
+          try {
+            val result = consumer.poll(frequency)
+            logger.debug(s"Consumer poll received ${result.count()} records from topic $topic")
```
Comment (jahhulbert-ccri) [inline geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: may be nice to have an ID here of some sort so you know which consumer is logging but not a big deal.

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  consumers.foreach(c => executor.execute(new ConsumerRunnable(c, frequency)))
+
+  logger.debug(s"Started ${consumers.length} consumer(s) on topic $topic")
+
+  override def close(): Unit = {
+    closed.set(true)
+    consumers.foreach(_.wakeup())
+    executor.shutdownNow()
+    executor.awaitTermination(1, TimeUnit.SECONDS)
```
Comment (jahhulbert-ccri) [inline geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: 1 sec may be too short for all consumers to close?

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+  protected val consumers: Seq[Consumer[Array[Byte], Array[Byte]]]
+  protected val topic: String
+  protected val frequency: Long
+
+  protected def consume(record: ConsumerRecord[Array[Byte], Array[Byte]]): Unit
+
+  private val closed = new AtomicBoolean(false)
+
+  private val executor = Executors.newFixedThreadPool(consumers.length)
```
Comment (jahhulbert-ccri) [inline geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: i almost want to have a dummy class for geomesa that extends this to make named threadpools so jstack instances show them easily.

**Comment by jahhulbert-ccri** [comment]
looks pretty good is this the fix for anthony's prob?

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala (hunk 1 above)
Comment (elahrvivaz) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: hmm, i don't see any identifiers for consumers - i could add a number to each thread?

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala (hunk 2 above)
Comment (elahrvivaz) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: how long do you think? i figured it doesn't really matter too much if the consumers don't actually close before this method returns

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala (hunk 3 above)
Comment (elahrvivaz) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: sure, add a ticket :)

**Comment by elahrvivaz** [comment]
it's a partial fix for anthony's issue - there's another PR that handles loading the whole topic without indexing that is the full solution

**Comment by jahhulbert-ccri** [comment]
ah yes i see

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala (hunk 1 above)
Comment (jahhulbert-ccri) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: yeah that's what i was thinking...we have that problem elsewhere...you have to override the threadfactory most likely. kinda gross.

Comment (jahhulbert-ccri) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: or use guava :)

Comment (elahrvivaz) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: haha, i could just add a val to the `ConsumerRunnable` when i create it - i'll go ahead and do that

Comment (jahhulbert-ccri) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: sounds good.

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala (hunk 2 above)
Comment (jahhulbert-ccri) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: yeah i guess it is not synchronous with the consumers closing...do you think it should be?

**Pull request merged**